www/authoring/questioneditor/v3/assetrequest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+    include_once '../../../config.php';
+    include_once 'includes/header.php';
+
+?>
+
+<div class="jumbotron section">
+    <div class="toolbar">
+        <ul class="list-inline">
+            <li data-toggle="tooltip" data-original-title="Visit the documentation"><a href="http://docs.learnosity.com/questioneditorapi/" title="Documentation"><span class="glyphicon glyphicon-book"></span></a></li>
+            <li data-toggle="tooltip" data-original-title="Toggle product overview box"><a href="#"><span class="glyphicon glyphicon-chevron-up jumbotron-toggle"></span></a></li>
+        </ul>
+    </div>
+    <div class="overview">
+        <h1>Question Editor API V3</h1>
+        <p>The Question Editor API can be extended to tie seamlessly in with your existing Digital Asset Management System.<p>
+    </div>
+</div>
+
+<!--
+********************************************************************
+*
+* Nav for different Question Editor API examples
+*
+********************************************************************
+-->
+<div class="section">
+    <!-- Container for the question editor api to load into -->
+    <script src="<?php echo $url_questioneditor_v3; ?>"></script>
+    <div class="learnosity-question-editor"></div>
+</div>
+<script>
+    /********************************************************************
+    *
+    * Set the different initialisation settings based off the
+    * example currently being requested
+    *
+    ********************************************************************/
+    var assetRequestFunction = function(mediaRequested, returnType, callback, attributes) {
+        if (mediaRequested === 'image') {
+            var $modal = $('.modal.img-upload'),
+            $images = $('.asset-img-gallery img'),
+            imgClickHandler = function () {
+                if (returnType === 'HTML') {
+                    callback('<img src="' + $(this).data('img') + '"/>');
+                } else {
+                    callback($(this).data('img'));
+                }
+                $modal.modal('hide');
+            };
+            $images.on('click', imgClickHandler);
+            $modal.modal({
+                backdrop: 'static'
+            }).on('hide', function () {
+                $images.off('click', imgClickHandler);
+            });
+        }
+    };
+
+    var initOptions = {
```

Comment:
There are two things that we should change here.

1. You need to add the _assetRequest_ attribute to the init JSON to make it work
`        "assetRequest": assetRequestFunction,
`
2. The imageclozeassociation question type is deprecated so we should use imageclozeassociationV2